lib/ex_unit/lib/ex_unit/capture_io.ex

```diff
@@ -228,23 +270,35 @@ defmodule ExUnit.CaptureIO do
     do_with_io(map_dev(device), options, fun)
   end
 
+  def with_io(pid, input, fun)
+      when is_pid(pid) and is_binary(input) and is_function(fun, 0) do
+    with_io(pid, [input: input], fun)
+  end
+
+  def with_io(pid, options, fun)
+      when is_pid(pid) and is_list(options) and is_function(fun, 0) do
+    do_with_io(pid, options, fun)
+  end
+
   defp map_dev(:stdio), do: :standard_io
   defp map_dev(:stderr), do: :standard_error
   defp map_dev(other), do: other
 
-  defp do_with_io(:standard_io, options, fun) do
+  defp do_with_io(device_or_pid, options, fun)
```

Comment:
I would have a separate clause for this instead:

```suggestion
  defp do_with_io(:standard_io, options, fun), do: do_with_io(self(), options, fun)
```

WDYT?

Reply:
Or maybe change this in map_dev. No worries, I will merge it and play with it locally.